src/pages/index.js

```diff
@@ -22,128 +23,261 @@ import {
+  submitHandler: handleAddCardFormSubmit,
+});
+
+const deleteCardPopup = new PopupWithForm({
+  popupSelector: "#delete-card-modal",
+  submitHandler: handleCardDelete,
+});
+
+const editAvatarPopup = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  submitHandler: handleAvatarFormSubmit,
+});
 
+// set popup event listeners
+imgagePopup.setEventListeners();
+profilePopup.setEventListeners();
+addCardPopup.setEventListeners();
+deleteCardPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
+
+//// Instantiate API ////
+const api = new Api({
+  baseUrl: baseUrl,
+  headers: {
+    authorization: token,
+    "Content-Type": "application/json",
+  },
+});
+
+//// Instantiate UserInfo ////
+const user = new UserInfo({ profileNameSelector, profileDescriptionSelector });
+
+api
+  .getUserInfo()
+  .then((result) => {
+    // process the result
+    user.setUserInfo({
+      name: result.name,
+      description: result.about,
+    });
+    profileAvatar.src = result.avatar;
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+//// Render initial cards ////
+const cardsSection = new Section(
+  { items: [], renderer: createCard },
+  gallerySelector
+);
+
+api
+  .getInitialCards()
+  .then((res) => {
+    if (res.length === 0) {
+      // If no cards are in the database
+      // add initialCards to server
+      initialCards.forEach((card) => {
+        api.addNewCard(card);
+      });
+    }
+
+    res.forEach((item) => {
+      // create new card
+      const newCard = createCard(item, openImageModal);
+      // Add new card to begining of card gallery
+      cardsSection.addItem(newCard);
+    });
+
+    cardsSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+//// Functions ////
 function fillProfileForm() {
   const userInfo = user.getUserInfo();
   profilePopup.setInputValues(userInfo);
+}
 
-  // profileNameInput.value = userInfo.name;
-  // profileDescriptionInput.value = userInfo.description;
+function openEditAvatarModal() {
+  editAvatarPopup.open();
 }
 
-// Button Handlers
 function openEditProfileModal() {
   fillProfileForm();
   formValidators[profileForm.getAttribute("id")].resetValidation();
 
   profilePopup.open();
 }
 
+function openImageModal(data) {
+  imgagePopup.open(data);
+}
+
 function openAddCardModal() {
   addCardPopup.open();
 }
 
-// Button Event Listeners
+function createCard(data) {
+  const cardElement = new Card(
+    data,
+    cardTemplate,
+    openImageModal,
+    openDeleteModal,
+    handleLikeButton
+  );
+
+  return cardElement.getView();
+}
+
+function openDeleteModal(currentCardId, currentCardElement) {
+  deleteCardPopup.setButtonText("Yes");
+  deleteCardPopup.open();
+  deleteCardPopup.currentCardId = currentCardId;
+  deleteCardPopup.currentCardElement = currentCardElement;
+}
+
+function handleCardDelete() {
+  deleteCardPopup.currentCardElement.remove();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error, the card should stay in the `DOM`.